signal_scanner_bot/env.py

```diff
@@ -45,4 +81,4 @@ def log_vars() -> None:
 
 SIGNAL_LOCK = Lock()
 
-STATE = _State()
+STATE = _State("signal_scanner_bot/.autoscanner-state-file")
```

Comment:
I dig this! But ideally we should have a default assigned in the `__init__` that points to this with an option to override if necessary. If we want to get fancy this could even be another `.env` value